### PR TITLE
gnat: 9 -> 11; update AdaCore packages

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

* Closes #143714.

* GNAT 11 is now required to build AdaCore's packages and we can also compile everything in nixpkgs so far using it, so we may as well upgrade to the latest version.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/ada/gnatcoll/bindings.nix [hunk 1]
```diff
@@ -68,7 +68,7 @@ stdenv.mkDerivation rec {
 
   buildPhase = ''
     runHook preBuild
-    python3 ${component}/setup.py build $buildFlags
+    python3 ${component}/setup.py build --prefix $out $buildFlags
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ada/gnatcoll/bindings.nix:71]:
```suggestion
    ${python.interpeter} ${component}/setup.py build --prefix $out $buildFlags
```

Comment (sternenseemann) [reply pkgs/development/libraries/ada/gnatcoll/bindings.nix]: Done.

**Comment by vcunat** [comment]
I don't know Ada stuff, but this sounds OK for 21.11.

**Comment by sternenseemann** [comment]
I think I'm gonna merge this early next week, if there are no concerns, given that Ada doesn't have a lot upstream (that is nixpkgs itself) stakeholders, I think I'd be struggling to find someone else to pull the trigger :p

Downstream concerns should be covered by release notes entry in principle.

**Pull request merged**